File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-12/08-pairsOfSongsWithTotalDurationDivisibleBy60.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        d = {}
        count = 0
        for t in time:
            if t % 60 == 0:
                count += d.get(0, 0)
            else:
                count += d.get(60 - t % 60, 0)
            d[t % 60] = d.get(t % 60, 0) + 1
        return count

File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-12/08-pairsOfSongsWithTotalDurationDivisibleBy60.java
class Solution {
    public int numPairsDivisibleBy60(int[] time) {
        Map<Integer, Integer> map = new HashMap<>();
        int count = 0;
        for (int t : time) {
            if (t % 60 == 0) {
                count += map.getOrDefault(0, 0);
            } else {
                count += map.getOrDefault(60 - t % 60, 0);
            }
            map.put(t % 60, map.getOrDefault(t % 60, 0) + 1);
        }
        return count;
    }
}
